src/main/java/org/folio/rest/impl/TemplateResourceImpl.java

```diff
@@ -4,40 +4,263 @@
 import io.vertx.core.Context;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.jaxrs.model.TemplateJson;
+import org.folio.rest.jaxrs.model.TemplatesCollectionJson;
 import org.folio.rest.jaxrs.resource.TemplateResource;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.Criteria.Limit;
+import org.folio.rest.persist.Criteria.Offset;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.folio.rest.tools.utils.TenantTool;
+import org.folio.rest.tools.utils.ValidationHelper;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSON;
 
 import javax.validation.constraints.NotNull;
 import javax.ws.rs.core.Response;
+import java.util.List;
 import java.util.Map;
+import java.util.UUID;
+
+import static io.vertx.core.Future.succeededFuture;
 
 public class TemplateResourceImpl implements TemplateResource {
 
-  private static final String TEMPLATE_STUB_PATH = "ramls/examples/template.sample";
-  private static final String CONTENT_TYPE_HEADER = "Content-Type";
-  private static final String APPLICATION_JSON = "application/json";
+  private static final String TEMPLATES_TABLE = "template";
+  private static final String TEMPLATES_ID_FIELD = "'id'";
+  public static final String TEMPLATE_SCHEMA_PATH = "ramls/template.json";
+  private static final String INTERNAL_ERROR = "Internal Server error";
+  private static final String POSTGRES_ERROR = "Error from PostgresClient: ";
+  private static final String INTERNAL_CONTEXT_ERROR = "Error running on vertx context: ";
+  private final Logger logger = LoggerFactory.getLogger("mod-template-engine");
 
   @Override
-  public void getTemplateByTemplateId(@NotNull String templateId, Map<String, String> okapiHeaders,
-                                      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    //TODO replace stub response
-    vertxContext.owner().fileSystem().readFile(TEMPLATE_STUB_PATH, event -> {
-      if (event.succeeded()) {
-        asyncResultHandler.handle(
-          Future.succeededFuture(
-            Response.ok(event.result().toString()).header(CONTENT_TYPE_HEADER, APPLICATION_JSON).build()
-          ));
+  public void postTemplate(
+    TemplateJson entity,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
```

Comment:
could you please format the code and mark parameters as final

Reply:
final is overhead for here. Thanks for code format remark